ext/gmp/gmp.c

```diff
@@ -862,6 +862,27 @@ static inline void _gmp_unary_opl(INTERNAL_FUNCTION_PARAMETERS, gmp_unary_opl_t
 }
 /* }}} */
 
+static bool gmp_verify_base(zend_long base)
+{
+	if (base && (base < 2 || base > GMP_MAX_BASE)) {
+		zend_argument_value_error(2, "must be between 2 and %d", GMP_MAX_BASE);
+		return false;
+	}
+
+	return true;
+}
+
+static int gmp_initialize_number(mpz_ptr gmp_number, const zend_string *arg_str, zend_long arg_l, zend_long base)
+{
+	if (arg_str) {
+		return convert_zstr_to_gmp(gmp_number, arg_str, base, 1);
+	} else {
+		mpz_set_si(gmp_number, arg_l);
+	}
+
+	return SUCCESS;
```

Comment:
suggestion: no need for the else block
```suggestion
	}
	mpz_set_si(gmp_number, arg_l);
	return SUCCESS;
```